libraries/AP_HAL_SITL/SITL_State.cpp

```diff
@@ -239,15 +239,24 @@ void SITL_State::_check_rc_input(void)
     case 8*2:
     case 16*2: {
         // a packet giving the receiver PWM inputs
-        uint8_t i;
-        for (i=0; i<size/2; i++) {
+        for (uint8_t i=0; i<size/2; i++) {
             // setup the pwm input for the RC channel inputs
             if (i < _sitl->state.rcin_chan_count) {
                 // we're using rc from simulator
                 continue;
             }
-            if (pwm_pkt.pwm[i] != 0) {
-                pwm_input[i] = pwm_pkt.pwm[i];
+            uint16_t pwm = pwm_pkt.pwm[i];
+            if (pwm != 0) {
+                if (_sitl->rc_fail == SITL::SITL::SITL_RCFail_Throttle950) {
+                    if (i == 2) {
+                        // set throttle (assumed to be on channel 3...)
+                        pwm = 950;
+                    } else {
+                        // centre all other inputs
+                        pwm = 1500;
```

Comment:
I wonder what happened if this wasn't centered... (we should definitely test that).

Reply:
We're emulating RC receiver bind values; this seems reasonable to me.

Reply:
But the test is for RC failsafe with low PWM and not all receivers center all other channels (actually some lower all channels which has been a problem before - and hence my suggestion of testing it).

Reply:
That would be another enumeration value :-)

This is for typical receiver bind values - having the throttle channel low and all others at trim.

I've never dealt with a receiver which does all-low-values.